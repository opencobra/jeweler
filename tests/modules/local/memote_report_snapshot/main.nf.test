nextflow_process {

    name 'Test Process MEMOTE_REPORT_SNAPSHOT'
    script 'modules/local/memote_report_snapshot/main.nf'
    process 'MEMOTE_REPORT_SNAPSHOT'
    tag 'modules'
    tag 'modules_local'
    tag 'memote_report_snapshot'

    test('Should succeed for a valid SBML document') {
        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of(
                    [[id: 'e_coli_core'], file('https://raw.githubusercontent.com/opencobra/jeweler/testdata/data/model/e_coli_core.xml.gz')]
                )
                """
            }
        }

        then {
            assert process.success

            with(process.out.report) {
                assert size() == 1

                with(get(0)) {
                    assert get(1).endsWith('html')

                    with(path(get(1)).readLines()) {
                        assert size() == 37
                        assert contains('  <title>MemoteReportApp</title>')
                    }
                }
            }

            // Reported versions should match snapshots.
            assert snapshot(process.out.versions.first()).match('versions')
        }
    }

    test('Should fail for an invalid SBML document') {
        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of(
                    [[id: 'e_coli_invalid'], file('https://raw.githubusercontent.com/opencobra/jeweler/testdata/data/model/e_coli_invalid.xml.gz')]
                )
                """
            }
        }

        then {
            assert process.failed
        }
    }
}
