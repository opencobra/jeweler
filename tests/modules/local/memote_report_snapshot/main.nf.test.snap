{
    "versions": {
        "content": [
            "versions.yml:md5,20fac3fdb773b01977b0c965461b3c3f"
        ],
        "timestamp": "2023-11-28T16:14:08.861044389"
    }
}